services_app/src/main/java/org/opendatakit/services/sync/actions/fragments/LoginFragment.java

```diff
@@ -173,27 +133,45 @@ private void findViewsAndAttachListeners(View view){
       btnAnonymousSignIn.setOnClickListener(onButtonClick);
       btnUserSignIn.setOnClickListener(onButtonClick);
       btnAuthenticateCredentials.setOnClickListener(onButtonClick);
-   }
 
-   private void updateUserInterface(){
-      props = ((IOdkAppPropertiesActivity) this.getActivity()).getProps();
-      userState = UserState.valueOf(props.getProperty(CommonToolProperties.KEY_CURRENT_USER_STATE));
+      loginViewModel=new ViewModelProvider(requireActivity()).get(LoginViewModel.class);
 
-      updateCommonInfo();
+      loginViewModel.getServerUrl().observe(getViewLifecycleOwner(), new Observer<String>() {
+         @Override
+         public void onChanged(String s) {
+            tvServerUrl.setText(s);
+            tvServerUrl.setPaintFlags(tvServerUrl.getPaintFlags() | Paint.UNDERLINE_TEXT_FLAG);
+         }
+      });
+
+      loginViewModel.getCurrentUserState().observe(getViewLifecycleOwner(), new Observer<UserState>() {
+         @Override
+         public void onChanged(UserState userState) {
+            if (userState == UserState.LOGGED_OUT) {
+               inLoggedOutState();
+            } else if (userState == UserState.ANONYMOUS) {
+               inAnonymousState();
+            } else {
+               inAuthenticatedState();
+            }
+         }
+      });
 
-      if (userState == UserState.LOGGED_OUT) {
-         inLoggedOutState();
-      } else if (userState == UserState.ANONYMOUS) {
-         inAnonymousState();
-      } else {
-         inAuthenticatedState();
-      }
-   }
+      loginViewModel.checkIsAnonymousSignInUsed().observe(getViewLifecycleOwner(), new Observer<Boolean>() {
+         @Override
+         public void onChanged(Boolean aBoolean) {
+            if(!aBoolean){
+               btnAnonymousSignIn.setEnabled(true);
+            }
```

Comment:
What happens when it is tapped again, it will still be enabled correct? I think it should disable it.